src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -0,0 +1,457 @@
+                        }
+
+                        // Output the erroneous token to aid user in debugging
+                        _fragments.Add(new FixedTextFragment(token));
+                        break;
+                }
+
+                start = rcurly + 1;
+            }
+
+
+            // Output any trailing plain text
+            if (start < pattern.Length)
+                _fragments.Add(new FixedTextFragment(pattern.Substring(start)));
+        }
+
+        /// <summary>
+        /// Get the display name for a method
+        /// </summary>
+        /// <param name="method">A MethodInfo</param>
+        /// <returns>The display name</returns>
+        public string GetDisplayName(MethodInfo method)
+        {
+            return GetDisplayName(method, null);
+        }
+
+        /// <summary>
+        /// Get the display name for a method with args
+        /// </summary>
+        /// <param name="method">A MethodInfo</param>
+        /// <param name="args">Argument list for the method</param>
+        /// <returns>The display name</returns>
+        public string GetDisplayName(MethodInfo method, object[] args)
+        {
+            var result = new StringBuilder();
+
+            foreach (var fragment in _fragments)
+                result.Append(fragment.GetText(method, args));
+
+            return result.ToString();
+        }
+
+        #region Nested Classes Representing Name Fragments
+
+        private abstract class NameFragment
+        {
+            private const string THREE_DOTS = "...";
+
+            public abstract string GetText(MethodInfo method, object[] args);
+
+            protected static void AppendGenericTypeNames(StringBuilder sb, MethodInfo method)
+            {
+                sb.Append("<");
+                int cnt = 0;
+                foreach (Type t in method.GetGenericArguments())
+                {
+                    if (cnt++ > 0) sb.Append(",");
+                    sb.Append(t.Name);
+                }
+                sb.Append(">");
+            }
+
+            protected static string GetDisplayString(object arg, int stringMax)
+            {
+                string display = arg == null
+                    ? "null"
+                    : Convert.ToString(arg, System.Globalization.CultureInfo.InvariantCulture);
+
+                if (arg is double)
+                {
+                    double d = (double)arg;
+
+                    if (double.IsNaN(d))
+                        display = "double.NaN";
+                    else if (double.IsPositiveInfinity(d))
+                        display = "double.PositiveInfinity";
+                    else if (double.IsNegativeInfinity(d))
+                        display = "double.NegativeInfinity";
+                    else if (d == double.MaxValue)
+                        display = "double.MaxValue";
+                    else if (d == double.MinValue)
+                        display = "double.MinValue";
```

Comment:
With your code above, wouldn't the int or double be indicated in the method name? `MyTest<int>(MaxValue)` and `MyTest<double>(MaxValue)`?

If not, it is just a minor suggestion :smile: